internal/impl/elasticsearchv8/output.go

```diff
@@ -0,0 +1,373 @@
+				Advanced().
+				Optional(),
+			service.NewBatchPolicyField(esFieldBatching),
+		).
+		Example("Updating Documents", "When updating documents, the request body should contain a combination of a `doc`, `upsert`, and/or `script` fields at the top level, this should be done via mapping processors. `doc` updates using a partial document, `script` performs an update using a scripting language such as the built in Painless language, and `upsert` updates an existing document or inserts a new one if it doesn’t exist. For more information on the structures and behaviors of these fields, please see the https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html[Elasticsearch Update API^]", `
+output:
+  processors:
+    - mapping: |
+        meta id = this.id
+        root.doc = this
+  opensearch:
+    urls: [ TODO ]
+    index: foo
+    id: ${! @id }
+    action: update
+`)
+}
+
+func init() {
+	err := service.RegisterBatchOutput("elasticsearch_v8", elasticsearchConfigSpec(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (out service.BatchOutput, batchPolicy service.BatchPolicy, maxInFlight int, err error) {
+			if maxInFlight, err = conf.FieldMaxInFlight(); err != nil {
+				return
+			}
+			if batchPolicy, err = conf.FieldBatchPolicy(esFieldBatching); err != nil {
+				return
+			}
+			out, err = outputFromParsed(conf, mgr)
+			return
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func outputFromParsed(pConf *service.ParsedConfig, mgr *service.Resources) (*esOutput, error) {
+	conf, err := esConfigFromParsed(pConf)
+	if err != nil {
+		return nil, err
+	}
+	return &esOutput{
+		log:  mgr.Logger(),
+		conf: conf,
+	}, nil
+}
+
+type esOutput struct {
+	log  *service.Logger
+	conf *esConfig
+
+	client *elasticsearch.TypedClient
+}
+
+func (e *esOutput) Connect(ctx context.Context) error {
+	if e.client != nil {
+		return nil
+	}
+
+	client, err := elasticsearch.NewTypedClient(e.conf.clientOpts)
+	if err != nil {
+		return err
+	}
+
+	e.client = client
+	return nil
+}
+
+func (e *esOutput) WriteBatch(ctx context.Context, batch service.MessageBatch) error {
+	bulkWriter := e.client.Bulk()
+
+	for i := range batch {
+		if err := e.addOpToBatch(bulkWriter, batch, i); err != nil {
+			return fmt.Errorf("adding operation to batch: %w", err)
+		}
+	}
+
+	result, err := bulkWriter.Do(ctx)
+	if err != nil {
+		return fmt.Errorf("sending bulk request: %w", err)
+	}
+
+	if result.Errors {
+		var batchErr *service.BatchError
+		for i, item := range result.Items {
+			for _, responseItem := range item {
+				if responseItem.Error != nil {
+					err := errors.New(*responseItem.Error.Reason)
+					if batchErr == nil {
+						batchErr = service.NewBatchError(batch, err)
+					}
+					batchErr.Failed(i, err)
+				}
+			}
+		}
+		return batchErr
+	}
+
+	// time.Duration is an int64 counting nanonseconds, result.Took is an int64
+	// counting milliseconds
+	tookDuration := time.Duration(result.Took * 1e6)
```

Comment:
this is serverside duration right?

also can we please do `result.Took * time.Millisecond`?

Reply:
Yes it's server side. That doesn't work, you can't multiply an `int64` and a `time.Duration`. Stuff like `time.Millisecond * 300` only works because Go's shapeshifting integer constant magic

Reply:
Yeah you could do this tho

```
time.Duration(result.Took)*time.Millsecond
```